pkg/distributor/distributor.go

```diff
@@ -326,6 +354,10 @@ func (d *Distributor) validateSeries(ts ingester_client.PreallocTimeseries, user
 		nil
 }
 
+func (d *Distributor) validateMetadata(m *ingester_client.MetricMetadata, userID string) error {
+	return validation.ValidateMetadata(d.limits, userID, m)
+}
+
```

Comment:
I'm not sure this method brings anything at the moment. Should we get rid of it?

Reply:
As you're going to work on this, let's move it to the next PR.